src/compas/scene/scene.py

```diff
@@ -118,8 +119,18 @@ def add(self, item, parent=None, **kwargs):
                     raise Exception("Object context should be the same as scene context: {} != {}".format(kwargs["context"], self.context))
                 del kwargs["context"]  # otherwist the SceneObject receives "context" twice, which results in an error
             sceneobject = SceneObject(item=item, context=self.context, **kwargs)  # type: ignore
+
+        self.datastore[str(item.guid)] = item
         super(Scene, self).add(sceneobject, parent=parent)
+
         return sceneobject
+    
+    def add_group(self, name=None, parent=None, **kwargs):
+        # type: (str, SceneObject | TreeNode | None, dict) -> Group
+        parent = parent or self.root
+        group = Group(name=name, **kwargs)
+        super(Scene, self).add(group, parent=parent)
+        return group
```

Comment:
Another reason we must have this function is that: `Group` class is not registered with an `item` and can not automatically reconstructed with correct extension class in a different context like `Viewer`. For example in Viewer we must overwrite this function to use `compas_viewer.scene.Group` instead of the original one